src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner - Pausing
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Pauses harvests. Can only be called by the owner or the manager.
+    function pause() external onlyOwnerOrManager {
+        _pause();
+    }
+
+    /// @notice Unpauses harvests. Can only be called by the owner.
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner - Config
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Updates the manager address. Can only be called by owner.
+    /// @param _manager Address of the new manager.
+    function setManager(address _manager) external onlyOwner {
+        address oldManager = manager;
+
+        manager = _manager;
+        emit ManagerUpdated(_manager, oldManager);
+    }
+
+    /// @notice Updates the keeper address. Can only be called by owner.
+    /// @param _keeper Address of the new keeper.
+    function setKeeper(address _keeper) external onlyOwner {
+        address oldKeeper = keeper;
+
+        keeper = _keeper;
+        emit KeeperUpdated(_keeper, oldKeeper);
+    }
+
+    /// @notice Updates the frequency at which rewards are processed by the keeper. Can only be called by owner.
+    /// @param _claimFrequency The new claim frequency in seconds.
+    function setClaimFrequency(uint256 _claimFrequency) external onlyOwner {
+        uint256 oldClaimFrequency = claimFrequency;
+
+        claimFrequency = _claimFrequency;
+        emit ClaimFrequencyUpdated(_claimFrequency, oldClaimFrequency);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Manager - Config
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Updates the current value for the minimum executable price (in bps as proportion of an oracle price)
+    ///         for a CRV to WETH swap. The value should be more than the minimum value. Can be called by the owner or
+    ///         the manager.
+    /// @param _minOutBpsCrvToWeth The new value in bps.
+    function setMinOutBpsCrvToWethVal(uint256 _minOutBpsCrvToWeth) external onlyOwnerOrManager {
+        if (_minOutBpsCrvToWeth > MAX_BPS) {
+            revert InvalidBps(_minOutBpsCrvToWeth);
+        }
+
+        uint256 minOutBpsCrvToWethMin = minOutBpsCrvToWeth.min;
+        if (_minOutBpsCrvToWeth < minOutBpsCrvToWethMin) {
+            revert LessThanBpsMin(_minOutBpsCrvToWeth, minOutBpsCrvToWethMin);
+        }
+
+        uint256 oldMinOutBpsCrvToWethVal = minOutBpsCrvToWeth.val;
+        minOutBpsCrvToWeth.val = _minOutBpsCrvToWeth;
+
+        emit MinOutBpsCrvToWethValUpdated(_minOutBpsCrvToWeth, oldMinOutBpsCrvToWethVal);
+    }
```

Comment:
Do we need a similar function for other bps'?

Reply:
i decided just to add one in the test, since all of them contains the same structure, only diff is the naming and purpose. but test would be identical, i think may be redundant

Reply:
initially i misunderstood the question, thinking that was referring to the test file, and replied that with one was enough, since otherwise will be testing the same functionalities. 

For the rest of variables is required to have setters, yes. I though of the approach of the mapping initially, but got bit ugly in the code. 

Something like reading this in swap methods:

```
bpsTkn2TknConfig[address(FXS)][address(FRAX)].val

or 

bpsTkn2TknConfig[address(WETH)][address(DAI)].val
```

But also is true that it can get ugly on the setters with many of them doing the same. Not strong opinion in either approach tbh

Reply:
mapping architecture introduced in 084a9d54d5f9a8f537e79730245ab6c0f9b58d42